compiler/noirc_frontend/src/hir/comptime/interpreter/builtin.rs

```diff
@@ -859,6 +860,39 @@ fn expr_as_bool(
     })
 }
 
+// fn as_comptime(self) -> Option<[Expr]>
+fn expr_as_comptime(
+    arguments: Vec<(Value, Location)>,
+    return_type: Type,
+    location: Location,
+) -> IResult<Value> {
+    expr_as(arguments, return_type, location, |expr| {
+        if let ExprValue::Expression(ExpressionKind::Comptime(block_expr, _)) = expr {
+            Some(block_expression_to_value(block_expr))
+        } else if let ExprValue::Statement(StatementKind::Comptime(statement)) = expr {
+            let typ = Type::Slice(Box::new(Type::Quoted(QuotedType::Expr)));
+
+            // comptime { ... } as a statement wraps a block expression,
+            // and in that case we return the block expression statements
+            // (comptime as a statement can also be comptime for, but in that case we'll
+            // return the for statement as a single expression)
+            if let StatementKind::Expression(Expression {
+                kind: ExpressionKind::Block(block_expr),
```

Comment:
Good idea. I added `use ExpressionKind::Block` and shortened a variable name and now it's a bit better. It didn't work with `Expression` because it collides with `struct Expression`.